### ISSUE TITLE
Model Configuration is wrongly restrictive

### ISSUE THREAD

**Issue by mikhovr** [triton-inference-server/server#4065]
**Description**
When I'm trying to serve ONNX model, saved with the following configuration of `dynamic_axes`:

```
dynamic_axes={
    "input_ids": {0: "batch_size", 1: "sequence"},
    "attention_mask": {0: "batch_size", 1: "sequence"},
    "logits":{0: "batch_size"},
}
```

with the following config:

```
platform: "onnxruntime_onnx"
max_batch_size: 8
input [
  {
    name: "input_ids"
    data_type: TYPE_INT64
    dims: [ -1 ]
  },
  {
    name: "attention_mask"
    data_type: TYPE_INT64
    dims: [ -1 ]
  }
]
output [
  {
    name: "logits"
    data_type: TYPE_FP32
    dims: [ -1, 25]
    label_filename: "labels.txt"
  }
]

instance_group [
  {
    count: {{ instances_count }}
    kind: KIND_GPU
  }
]
```

TIS doesn't load this model with error:
```
the model expects 3 dimensions (shape [-1,256,25]) but the model configuration specifies 3 dimensions (an initial batch dimension because max_batch_size > 0 followed by the explicit tensor shape, making complete shape [-1,-1,25])
```

Isn't tensor shape `[-1,-1,25]` in configuration compatible with `[-1,256,25]` from ONNX signature?

**Triton Information**
Container `nvcr.io/nvidia/tritonserver:21.06-py3`

**Comment by wanghaohw** [comment]
There is only on dimension is dynamic```"logits":{0: "batch_size"},```，you can set ```"logits":{0: "batch_size",1: "para"},``` and have a try.

**Comment by mikhovr** [comment]
Yes, it's working with `"logits":{0: "batch_size", 1: "sequence"}`. 
Also, it's working with 
```
output [
  {
    name: "logits"
    data_type: TYPE_FP32
    dims: [ 256, 25]
    label_filename: "labels.txt"
  }
]
```
in Model Configuration.

The problem is that some of the production models are saved with the only `"logits":{0: "batch_size"}` and TIS config for them is generated from the template, thus making impossible using one config for the two types of models.

**Comment by dzier** [comment]
That is by design, Triton expects a model config for each type of model, so if you have two different types, you will need two different model configs.

**Comment by justcodew** [comment]
![image](https://user-images.githubusercontent.com/107862224/188622975-639f9999-87f7-4645-b1a9-e1f7342f8f2b.png)
i come across some error like this
`
Invalid argument: model 'smoke_plan', tensor 'data': the model expects 4 dimensions (shape [-1,3,384,544]) but the model configuration specifies 4 dimensions (an initial batch dimension because max_batch_size > 0 followed by the explicit tensor shape, making complete shape [-1,3,544,384])
`

**Comment by GuanLuo** [comment]
As you can see, the model config that you provided swaps the 3rd and 4th dimension, which is not what your model expects. You should fix the model config

**Comment by osuzdalevZeiss** [comment]
Get a similar error:
`Invalid argument: model 'groot', tensor 'boxes': the model expects 2 dimensions (shape [91,4]) but the model configuration specifies 2 dimensions (shape [-1,-1])`

Saved the model with torch.onnx.export and the following dynamic_axes dictionary:
```
dynamic_axes={"model_input": {0: "batch_size", 1: "channels", 2: "width", 3: "height"},
                                            "masks": {0: "batch_size", 1: "width", 2: "height"},
                                            "scores": {0: "scores_size"},
                                            "labels": {0: "labels_size"},
                                            "boxes": {0: "boxes_size", 1: "boxes_param"}}
```

Here is my config file:
```
name: "groot"
platform: "onnxruntime_onnx"
max_batch_size: 0
instance_group [
    {
      count: 1
      kind: KIND_GPU
    }
]

input [
  {
    name: "model_input"
    data_type: TYPE_FP32
    dims: [-1,-1,-1,-1]
  }
]
output [
  {
    name: "boxes"
    data_type: TYPE_FP32
    dims: [-1, -1]
  },
    {
    name: "masks"
    data_type: TYPE_UINT8
    dims: [-1]
  },
    {
    name: "labels"
    data_type: TYPE_UINT64
    dims: [-1]
  },
    {
    name: "scores"
    data_type: TYPE_FP32
    dims: [-1]
  }
]
```

Can I get any help as to what I am doing wrong?

**Comment by GuanLuo** [comment]
Can you open the model in visualize tools like [Netron](https://netron.app/) to check the I/O expected by the model?

**Comment by GuanLuo** [comment]
And please create a separate issue if follow-up is needed because closed issue is hard to track.

**Comment by osuzdalevZeiss** [comment]
Hi, thanks for answering!
I created a new issue: https://github.com/triton-inference-server/server/issues/5834#issue-1721174398
